app/tests/unit/containers/Forms/ProductField.test.tsx

```diff
@@ -0,0 +1,397 @@
+import React from 'react';
+import {shallow} from 'enzyme';
+import {ProductFieldComponent} from 'containers/Forms/ProductField';
+import productionSchema from 'schema/data/prod/form_json/production.json';
+import {JSONSchema7} from 'json-schema';
+import {getDefaultRegistry} from '@rjsf/core/dist/cjs/utils';
+import FormArrayFieldTemplate from 'containers/Forms/FormArrayFieldTemplate';
+import FormObjectFieldTemplate from 'containers/Forms/FormObjectFieldTemplate';
+import FormFieldTemplate from 'containers/Forms/FormFieldTemplate';
+import {ProductField_query} from '__generated__/ProductField_query.graphql';
+import {ProductField_naicsCode} from '__generated__/ProductField_naicsCode.graphql';
+
+describe('The ProductionFields Component with published product matched to a naics code', () => {
+  const query: ProductField_query = {
+    ' $refType': 'ProductField_query',
+    allProducts: {
+      edges: [
+        {
+          node: {
+            rowId: 1,
+            units: 'bar',
+            productState: 'PUBLISHED',
+            requiresEmissionAllocation: true,
+            requiresProductAmount: true,
+            isEnergyProduct: false,
+            addPurchasedElectricityEmissions: false,
+            addPurchasedHeatEmissions: false,
+            subtractExportedElectricityEmissions: false,
+            subtractExportedHeatEmissions: false,
+            subtractGeneratedElectricityEmissions: false,
+            subtractGeneratedHeatEmissions: false,
+            addEmissionsFromEios: false
+          }
+        }
+      ]
+    }
+  };
+  const naicsCode: ProductField_naicsCode = {
+    ' $refType': 'ProductField_naicsCode',
+    allProductsByNaicsCode: {
+      edges: [
+        {
+          node: {
+            rowId: 1,
+            productState: 'PUBLISHED'
+          }
+        }
+      ]
+    }
+  };
+
+  const idSchema: any = {$id: 'product_0'};
+  const props = {
+    schema: productionSchema.schema.definitions.product as JSONSchema7,
+    uiSchema: productionSchema.uiSchema,
+    idSchema,
+    autofocus: false,
+    disabled: false,
+    errorSchema: null,
+    formContext: {},
+    readonly: false,
+    required: false,
+    registry: {
+      ...getDefaultRegistry(),
+      ArrayFieldTemplate: FormArrayFieldTemplate,
+      ObjectFieldTemplate: FormObjectFieldTemplate,
+      FieldTemplate: FormFieldTemplate
+    } as any,
+
+    name: 'product',
+    query,
+    naicsCode,
+    formData: {
+      productRowId: 1,
+      productUnits: 'bar',
+      productEmissions: 123,
+      productAmount: 456,
+      requiresEmissionAllocation: true
+    },
+    onChange: jest.fn(),
+    onBlur: jest.fn()
```

Comment:
there is a `createDefaultJsonSchemaFormProps()` method in [json-schema-utils.ts](https://github.com/bcgov/cas-ciip-portal/blob/feat/filter-products-by-naics/app/tests/json-schema-utils.ts) that makes these test setups a lot shorter

I don't think a reference to the production schema is needed there since we only do a `shallow` render?

Reply:
nice, I'll give that a shot. It was a huge chunk of code for each test!